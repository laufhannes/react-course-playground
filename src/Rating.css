.Rating-stars {
  font-size: 76px;
  color: gold;
  user-select: none;
}

.Rating-stars span:hover {
  cursor: pointer;
}