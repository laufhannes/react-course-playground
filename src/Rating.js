import React, { PureComponent } from "react";
import "./Rating.css";

class Rating extends PureComponent {
  // As this component is a PureComponent, `render()` will only be called if `newProps.stars != oldProps.stars`
  render() {
    const starData = [];

    for (let i = 1; i <= this.props.maxStars; ++i) {
      starData.push({
        id: i,
        active: i <= this.props.stars
      });
    }

    return (
      <div className="Rating-stars">
        {starData.map(star => (
          <span
            key={star.id.toString()}
            onClick={evt => {
              this.props.onRatingChange(
                1 === star.id && 1 === this.props.stars ? 0 : star.id
              );
            }}
          >
            {star.active ? this.props.fullStar : this.props.emptyStar}
          </span>
        ))}
      </div>
    );
  }
}

Rating.defaultProps = {
  maxStars: 5,
  fullStar: "★",
  emptyStar: "☆",
  onRatingChange: () => {}
};

export default Rating;
