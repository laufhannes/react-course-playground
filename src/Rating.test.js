import React from 'react';
import Rating from './Rating';
import renderer from 'react-test-renderer';

it('renders-correctly', () => {
  const tree = renderer
    .create(<Rating stars={4} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
})

it('renders-correctly with 3 stars', () => {
  const tree = renderer
    .create(<Rating stars={3} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
})
