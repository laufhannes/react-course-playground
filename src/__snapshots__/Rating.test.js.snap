// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders-correctly 1`] = `
<div
  className="Rating-stars"
>
  <span
    onClick={[Function]}
  >
    ★
  </span>
  <span
    onClick={[Function]}
  >
    ★
  </span>
  <span
    onClick={[Function]}
  >
    ★
  </span>
  <span
    onClick={[Function]}
  >
    ★
  </span>
  <span
    onClick={[Function]}
  >
    ☆
  </span>
</div>
`;

exports[`renders-correctly with 3 stars 1`] = `
<div
  className="Rating-stars"
>
  <span
    onClick={[Function]}
  >
    ★
  </span>
  <span
    onClick={[Function]}
  >
    ★
  </span>
  <span
    onClick={[Function]}
  >
    ★
  </span>
  <span
    onClick={[Function]}
  >
    ☆
  </span>
  <span
    onClick={[Function]}
  >
    ☆
  </span>
</div>
`;
